src/include/hip/rpp_hip_common.hpp

```diff
@@ -72,4 +114,441 @@ inline RppStatus generate_gaussian_kernel_gpu(Rpp32f stdDev, Rpp32f* kernel, Rpp
+{
+    return (float)((src >> 8) & 0xFF);
+}
+
+__device__ __forceinline__ float rpp_hip_unpack2(uint src)
+{
+    return (float)((src >> 16) & 0xFF);
+}
+
+__device__ __forceinline__ float rpp_hip_unpack3(uint src)
+{
+    return (float)((src >> 24) & 0xFF);
+}
+
+__device__ __forceinline__ float4 rpp_hip_unpack(uint src)
+{
+    return make_float4(rpp_hip_unpack0(src), rpp_hip_unpack1(src), rpp_hip_unpack2(src), rpp_hip_unpack3(src));
+}
+
+__device__ __forceinline__ float rpp_hip_unpack0(int src)
+{
+    return (float)(src & 0xFF);
+}
+
+__device__ __forceinline__ float rpp_hip_unpack1(int src)
+{
+    return (float)((src >> 8) & 0xFF);
+}
+
+__device__ __forceinline__ float rpp_hip_unpack2(int src)
+{
+    return (float)((src >> 16) & 0xFF);
+}
+
+__device__ __forceinline__ float rpp_hip_unpack3(int src)
+{
+    return (float)((src >> 24) & 0xFF);
+}
+
+__device__ __forceinline__ float4 rpp_hip_unpack_from_i8(int src)
+{
+    return make_float4(rpp_hip_unpack0(src), rpp_hip_unpack1(src), rpp_hip_unpack2(src), rpp_hip_unpack3(src));
+}
+
+// U8 loads and stores without layout toggle (8 U8 pixels)
+
+__device__ __forceinline__ void rpp_hip_load8_and_unpack_to_float8(uchar *srcPtr, uint srcIdx, d_float8 *src_f8)
+{
+    uint2 src = *((uint2 *)(&srcPtr[srcIdx]));
+    src_f8->x = rpp_hip_unpack(src.x);
+    src_f8->y = rpp_hip_unpack(src.y);
+}
+
+__device__ __forceinline__ void rpp_hip_pack_float8_and_store8(uchar *dstPtr, uint dstIdx, d_float8 *dst_f8)
+{
+    uint2 dst;
+    dst.x = rpp_hip_pack(dst_f8->x);
+    dst.y = rpp_hip_pack(dst_f8->y);
+    *((uint2 *)(&dstPtr[dstIdx])) = dst;
+}
+
+// F32 loads and stores without layout toggle (8 F32 pixels)
+
+__device__ __forceinline__ void rpp_hip_load8_and_unpack_to_float8(float *srcPtr, uint srcIdx, d_float8 *src_f8)
+{
+    *src_f8 = *((d_float8 *)(&srcPtr[srcIdx]));
+}
+
+__device__ __forceinline__ void rpp_hip_pack_float8_and_store8(float *dstPtr, uint dstIdx, d_float8 *dst_f8)
+{
+    *((d_float8 *)(&dstPtr[dstIdx])) = *dst_f8;
+}
+
+// I8 loads and stores without layout toggle (8 I8 pixels)
+
+__device__ __forceinline__ void rpp_hip_load8_and_unpack_to_float8(signed char *srcPtr, uint srcIdx, d_float8 *src_f8)
+{
+    uint2 src = *((uint2 *)(&srcPtr[srcIdx]));
+    src_f8->x = rpp_hip_unpack(src.x);
+    src_f8->y = rpp_hip_unpack(src.y);
+}
+
+__device__ __forceinline__ void rpp_hip_pack_float8_and_store8(signed char *dstPtr, uint dstIdx, d_float8 *dst_f8)
+{
+    uint2 dst;
+    dst.x = rpp_hip_pack(dst_f8->x);
+    dst.y = rpp_hip_pack(dst_f8->y);
+    *((uint2 *)(&dstPtr[dstIdx])) = dst;
+}
+
+// F16 loads and stores without layout toggle (8 F16 pixels)
+
+__device__ __forceinline__ void rpp_hip_load8_and_unpack_to_float8(half *srcPtr, uint srcIdx, d_float8 *src_f8)
+{
+    d_half8 src_h8;
+    src_h8 = *((d_half8 *)(&srcPtr[srcIdx]));
+
+    *(float2 *)&(src_f8->x) = __half22float2(src_h8.x.x);
+    *((float2 *)&(src_f8->x) + 1) = __half22float2(src_h8.x.y);
```

Comment:
don't use +1 here. First, create a temp float 4 and fill it based on __half22float2(src_h8.x.x)/__half22float2(src_h8.x.y) and the assign that float4 to the src_f8->x. do the same for src_f8->y

Reply:
Done